libraries/Microsoft.Bot.Builder.Core.Extensions/FileTranscriptStore.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Core.Extensions
+{
+    /// <summary>
+    /// The file transcript store stores transcripts in file system with each activity as a file.
+    /// </summary>
+    public class FileTranscriptStore : ITranscriptStore
+    {
+        private static JsonSerializerSettings jsonSettings = new JsonSerializerSettings()
+        {
+            Formatting = Formatting.Indented,
+            NullValueHandling = NullValueHandling.Ignore
+        };
+        protected string rootFolder;
+
+        public FileTranscriptStore(string folder)
+        {
+            this.rootFolder = folder ?? throw new ArgumentNullException("Missing folder");
+        }
+
+        /// <summary>
+        /// Log an activity to the transcript
+        /// </summary>
+        /// <param name="activity">activity to log</param>
+        /// <returns></returns>
+        public async Task LogActivity(IActivity activity)
+        {
+            if (activity == null)
+                throw new ArgumentNullException("activity cannot be null for LogActivity()");
+
+            var activityPath = GetActivityPath(activity);
+
+            try
+            {
+                File.WriteAllText(activityPath, JsonConvert.SerializeObject(activity, jsonSettings));
+            }
+            catch (DirectoryNotFoundException)
+            {
+                Directory.CreateDirectory(Path.GetDirectoryName(activityPath));
+                File.WriteAllText(activityPath, JsonConvert.SerializeObject(activity, jsonSettings));
```

Comment:
>JsonConvert.SerializeObject [](start = 48, length = 27)

You'll always serialize twice on the first activity for a conversation. Since you're not catching serialization errors anyway seems like you should serialize the activity once outside the scope of your try { } catch() { } block